cider/exec/nextgen/operators/ColumnToRowNode.cpp

```diff
@@ -45,13 +45,46 @@ class ColumnReader {
       case kDOUBLE:
         readFixSizedTypeCol();
         break;
+      case kVARCHAR:
+      case kCHAR:
+      case kTEXT:
+        readVariableSizeTypeCol();
+        break;
       default:
         LOG(FATAL) << "Unsupported data type in ColumnReader: "
                    << expr_->get_type_info().get_type_name();
     }
   }
 
  private:
+  void readVariableSizeTypeCol() {
+    auto&& [batch, buffers] = context_.getArrowArrayValues(expr_->getLocalIndex());
+    utils::VarSizeJITExprValue varsize_values(buffers);
+
+    auto& func = batch->getParentJITFunction();
+    // offset buffer
+    auto offset_pointer =
+        varsize_values.getLength()->castPointerSubType(JITTypeTag::INT32);
+    auto len = offset_pointer[index_ + 1] - offset_pointer[index_];
+    auto cur_offset = offset_pointer[index_];
+    // data buffer
+    auto value_pointer = varsize_values.getValue()->castPointerSubType(JITTypeTag::INT8);
+    auto row_data = value_pointer + cur_offset;  // still char*
+
+    if (expr_->get_type_info().get_notnull()) {
+      expr_->set_expr_value<JITExprValueType::ROW>(nullptr, len, row_data);
```

Comment:
Should set NULL to `func.createConstant(JITTypeTag::BOOL, false)` instead of `nullptr` here.

Reply:
updated.